File: VB/MenuExample/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports MenuExample.Models

Namespace MenuExample.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			' DXCOMMENT: Pass a data model for GridView
			Return View()
		End Function

		Public Function GridViewPartialView() As ActionResult
			' DXCOMMENT: Pass a data model for GridView in the PartialView method's second parameter
			Return PartialView("GridViewPartialView", NorthwindDataProvider.GetCustomers())
		End Function
		Public Function CallbackPanelPartial(ByVal view As String) As ActionResult
			ViewData("view") = view
			Return PartialView()
		End Function
		Public Function GridViewCategoriesPartial() As ActionResult

			Return PartialView(NorthwindDataProvider.DB.Categories)
		End Function
		Public Function GridViewProductsPartial() As ActionResult
			Return PartialView(NorthwindDataProvider.DB.Products)
		End Function
	End Class
End Namespace

File: CS/MenuExample/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using MenuExample.Models;

namespace MenuExample.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
			// DXCOMMENT: Pass a data model for GridView
            return View();	
        }
		
		public ActionResult GridViewPartialView() 
		{
            // DXCOMMENT: Pass a data model for GridView in the PartialView method's second parameter
			return PartialView("GridViewPartialView", NorthwindDataProvider.GetCustomers());
        }
        public ActionResult CallbackPanelPartial(string view)
        {
            ViewData["view"] = view;   
            return PartialView();
        }
        public ActionResult GridViewCategoriesPartial() {

            return PartialView(NorthwindDataProvider.DB.Categories);
        }
        public ActionResult GridViewProductsPartial()
        {
            return PartialView(NorthwindDataProvider.DB.Products);
        }
	}
}